http/middleware/compression.go

```diff
@@ -0,0 +1,55 @@
+package middleware
+
+import (
+	"strings"
+
+	gbrotli "github.com/anargu/gin-brotli"
+	"github.com/andybalholm/brotli"
+	"github.com/gin-contrib/gzip"
+	"github.com/gin-gonic/gin"
```

Comment:
Q: Can we remove the gin dependency totally?

Reply:
Yes it should be possible I think. I found  [this package](https://github.com/nytimes/gziphandler) but it requires an http.Handler as input. Any clue how to use this here instead of the gin one?

Reply:
We may need to read the source code to understand how to implement Gzip.

By the way, I missed one thing, we can use Gzip by nginx, is it necessary to implement Gzip through the framework? Most time, we need to use Nginx to make a request forwarding. cc @Marian0

Reply:
My 2 cents is that it is something worth it to have in the framework. There was a time until ~2016 that it was preferred to always expose go services being a proxy, (such as nginx),. But then companies started to expose their go services directly and I believe not it is actually safe to do that. Ofc you do not get the benefits of a true proxy but compression is a nice to have for such cases.